### ISSUE TITLE
Resolve dependency cycle between ModalManager and withModalManager HOC

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#245]
# Feature Request

## Description
When updating to ESLint 5 I noticed we have a dependency cycle in [ModalManager.jsx where we import the withModalManager HOC](https://github.com/cerner/terra-framework/blob/master/packages/terra-modal-manager/src/ModalManager.jsx#L11). Then inside of [withModalManager.jsx, we import ModalManager](https://github.com/cerner/terra-framework/blob/master/packages/terra-modal-manager/src/withModalManager.jsx).

This hasn't seemed to causes some issues but it would probably be best to decouple this dependency cycle. 

I'd recommend we drop the import of withModalManager from the ModalManager.jsx file. This would be non-passive as people may be interfacing directly with ModalManager.jsx to get the withModalManager HOC. With this change, they would need to interface with withModalManager.jsx to get the HOC.

**Comment by tbiethman** [comment]
We could also make a new file that imports both and make it the new main file:

```jsx
// NewMain.js
import ModalManager from './ModalManager';
import withModalManager from './withModalManager';

export default ModalManager;
export { withModalManager };

```

That should at least be passive.

**Comment by bjankord** [comment]
Resolved in #318
